utils.go

```diff
@@ -82,17 +83,39 @@ func GetRootlessRuntimeDir(rootlessUID int) (string, error) {
 	return path, nil
 }
 
+func getInitCommand() (string, error) {
+	file, err := os.Open("/proc/1/comm")
+	if err != nil {
+		return "", err
+	}
+	defer file.Close()
+	buf, err := ioutil.ReadAll(file)
+	if err != nil {
+		return "", err
+	}
+	return string(buf), nil
+}
+
 func getRootlessRuntimeDir(rootlessUID int) (string, error) {
 	runtimeDir, err := homedir.GetRuntimeDir()
 	if err == nil {
 		return runtimeDir, nil
 	}
-	tmpDir := fmt.Sprintf("/run/user/%d", rootlessUID)
+	initCommand, err := getInitCommand()
+	var tmpDir string
+	if err == nil && initCommand == "systemd" {
```

Comment:
let's invert the logic here and default to systemd: `if err != nil || initCommand == "systemd"`

Reply:
done